### ISSUE TITLE
App doesn't start with "Unable to find zbar shared library" error

### ISSUE THREAD

**Issue by fuzzmz** [bilelmoussaoui/Authenticator#58]
**Reporter info**

 ```
 * Distribution - Ubuntu 17.10
 * Gtk+ 3.0 version - 3.22.25-0ubuntu0.1
 * Desktop environment - GNOME Shell 3.26.2
 ```

**Authenticator version**
Updating: com.github.bilelmoussaoui.Authenticator/x86_64/stable from flathub
No updates.
Updating: com.github.bilelmoussaoui.Authenticator.Locale/x86_64/stable from flathub
No updates

**Actual issue**
Authenticator crashes with following stack trace:

```** (authenticator:3): WARNING **: Couldn't connect to accessibility bus: Failed to connect to socket /tmp/dbus-JJCMcEcLaG: Connection refused
Traceback (most recent call last):
  File "/app/bin/authenticator", line 31, in <module>
    from Authenticator import Application
  File "/app/lib/python3.5/site-packages/Authenticator/__init__.py", line 19, in <module>
    from .application import Application
  File "/app/lib/python3.5/site-packages/Authenticator/application.py", line 24, in <module>
    from .widgets import Window, AboutDialog
  File "/app/lib/python3.5/site-packages/Authenticator/widgets/__init__.py", line 19, in <module>
    from .window import Window
  File "/app/lib/python3.5/site-packages/Authenticator/widgets/window.py", line 22, in <module>
    from ..models import Logger, Settings, Database
  File "/app/lib/python3.5/site-packages/Authenticator/models/__init__.py", line 20, in <module>
    from .qr_reader import QRReader
  File "/app/lib/python3.5/site-packages/Authenticator/models/qr_reader.py", line 24, in <module>
    from pyzbar.pyzbar import decode
  File "/app/lib/python3.5/site-packages/pyzbar-0.1.4-py3.5.egg/pyzbar/pyzbar.py", line 8, in <module>
    from .wrapper import (
  File "/app/lib/python3.5/site-packages/pyzbar-0.1.4-py3.5.egg/pyzbar/wrapper.py", line 166, in <module>
    c_uint_p,    # minor
  File "/app/lib/python3.5/site-packages/pyzbar-0.1.4-py3.5.egg/pyzbar/wrapper.py", line 159, in zbar_function
    return prototype((fname, load_libzbar()))
  File "/app/lib/python3.5/site-packages/pyzbar-0.1.4-py3.5.egg/pyzbar/wrapper.py", line 135, in load_libzbar
    raise ImportError('Unable to find zbar shared library')
ImportError: Unable to find zbar shared library
```

**Steps to reproduce (if you knew)**
`flatpak install flathub com.github.bilelmoussaoui.Authenticator`
`flatpak run com.github.bilelmoussaoui.Authenticator`

**Other Note (feature-request, question, etc...)**
Tried manually installing `libzbar-dev` but issue still reproduces.

**Comment by bilelmoussaoui** [comment]
Which arch are you running the application on? on x64_86 the app does run correctly
Will investigate if you can give me some more details

**Comment by reblws** [comment]
I'm having the same issue:
```
* Distribution - Debian 9 (Stretch)
* Gtk+ 3.0 version - libgtk-3-0:amd64 - 3.22.11-1 
* Desktop environment - GNOME Shell 3.22.2
```

**Comment by fuzzmz** [comment]
@bilelmoussaoui, running on x86_64.

```
flatpak info com.github.bilelmoussaoui.Authenticator    


app/com.github.bilelmoussaoui.Authenticator/x86_64/stable flathub 37bab1810ffcd2e821aa035eaeba9ab0f1a595c1c405be3622a375e112a69c73
```

**Comment by jupiter98** [comment]
I was running Linux Mint 18.3 and I was able to install Authenticator and it worked perfectly.
Then I changed to running the latest MX Linux 17.1 which is based on Debian 9.4 version
and I am also seeing the same issue that it won't run and shows

  File "/app/lib/python3.5/site-packages/pyzbar-0.1.4-py3.5.egg/pyzbar/wrapper.py", line 135, in load_libzbar
    raise ImportError('Unable to find zbar shared library')
ImportError: Unable to find zbar shared library

If a company wants to join and add their company to the list of around 290 in Authenticator so far what do they need to do?
Thanks for a really great app!

**Comment by jupiter98** [comment]
I was running x64_86 on both Linux versions.

**Comment by bilelmoussaoui** [comment]
So the issue happens only with Debian 9.4 or distros that are based on it?

**Comment by bilelmoussaoui** [comment]
@jupiter98 I use the database of https://github.com/2factorauth/twofactorauth that I convert from yaml to json to keep only the data that interests the user which means the company name!

**Comment by jupiter98** [comment]
Sorry I can't say if the issue only occurs on Debian 9.4 and its distros as I have only tried to install the Authenticator on Linux Mint 18.3 and MX Linux 17.1

**Comment by jupiter98** [comment]
Sorry I can't say if the issue only occurs on Debian 9.4 and its distros as I have only tried to install the Authenticator on Linux Mint 18.3 and MX Linux 17.1

**Comment by jupiter98** [comment]
I installed wheel and zbar but
pip install zbar
seemed to install zbar so that its associated with Python 2.7
and the traceback in the error is showing Python 3.5
and the same error is still occurring.

**Comment by reblws** [comment]
@bilelmoussaoui it does seem like this issue only happens with Debian-based distros. I originally thought there was a bug somewhere in the flatpak installation process that fails to install the zbar library to the flatpak lib directory but here's what I see in `/var/lib/flatpak/app/com.github.bilelmoussaoui.Authenticator/x86_64/stable/37bab1810ffcd2e821aa035eaeba9ab0f1a595c1c405be3622a375e112a69c73/files/lib`:

```
drwxr-xr-x 6 root root 4.0K Dec 31  1969 .
drwxr-xr-x 6 root root 4.0K Apr  8 03:33 ..
drwxr-xr-x 2 root root 4.0K Dec 31  1969 debug
-rwxr--r-- 2 root root 1.2M Dec 31  1969 libzbar.a
-rwxr-xr-x 2 root root  955 Dec 31  1969 libzbar.la
lrwxrwxrwx 1 root root   16 Apr  8 03:33 libzbar.so -> libzbar.so.0.2.0
lrwxrwxrwx 1 root root   16 Apr  8 03:33 libzbar.so.0 -> libzbar.so.0.2.0
-rwxr-xr-x 2 root root 156K Dec 31  1969 libzbar.so.0.2.0
drwxr-xr-x 2 root root 4.0K Dec 31  1969 pkgconfig
drwxr-xr-x 3 root root 4.0K Dec 31  1969 python2.7
drwxr-xr-x 3 root root 4.0K Dec 31  1969 python3.5
```

**Comment by jupiter98** [comment]
Given that Ubuntu is based on Debian and Linux Mint is based on Ubuntu would you have any ideas what
they changed which has solved this issue?

**Comment by bilelmoussaoui** [comment]
I think it's due to the way I'm building zbar. I'm going to take a look as soon as I can.

**Comment by bilelmoussaoui** [comment]
If anyone is familiar with Flatpak and has Debian installed, they can try to check out the flatpak manifest from here
https://github.com/flathub/com.github.bilelmoussaoui.Authenticator/

**Comment by fuzzmz** [comment]
I'll try flatpak-building using that manifest and see what's up. Currently during flatpak-build it fails with 

```
Building module pyotp in /home/fuzzmz/test/com.github.bilelmoussaoui.Authenticator/.flatpak-builder/build/pyotp-1
========================================================================
Error: module pyotp: Invalid buildsystem: "simple"```

I'll update when I have progress.

**Comment by bilelmoussaoui** [comment]
Which version of flatpak-builder do you have?

**Comment by jupiter98** [comment]
Have you noticed that the page
https://flathub.org/apps/details/com.github.bilelmoussaoui.Authenticator
only has
$ flatpak install flathub
and
$ flatpak run
as the Command line instructions?

I installed Opensuse Leap 42.3 today, updated it, then installed the latest flatpak-builder, then the Authenticator, and I can say now that the error is not isolated to Debian and its Distros.
I saw exactly the same error again when I tried to run the Authenticator.

**Comment by bilelmoussaoui** [comment]
Hmm, I will try to fix this tonight. Really sorry for the long delay. Working on a new version of the application that should add a lot of improvements!

**Comment by bilelmoussaoui** [comment]
I have installed the latest stable debian version out there. Installed flatpak. But they still ship a really old version :'(

**Comment by bilelmoussaoui** [comment]
So I have decided to compile zbar on Debian, got no errors during the compile time and the application works without any issues... 
The issues you guys are seeing comes from https://github.com/NaturalHistoryMuseum/pyzbar/blob/master/pyzbar/zbar_library.py#L63
If you do try to run 
```

from ctypes.util import find_library

print(find_library('zbar'))
```
everything works as expected. I'm going to see if any of the flatpak devs have an idea why is this happening

**Comment by bilelmoussaoui** [comment]
I have figured out how to fix this issue. It will be fixed with the next version. I will try to release by the 13/06 after my finals.

**Comment by bilelmoussaoui** [comment]
Fixed on the 0.2.2 release.
